pkg/driver/node.go

```diff
@@ -149,8 +149,25 @@ func (d *Driver) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolu
 		return nil, status.Error(codes.InvalidArgument, "Staging target not provided")
 	}
 
+	// Check if target directory is a mount point. GetMountRefs
+	// finds all mount references to the source, returns a list of
+	// mountpoints.
+	refs, err := d.mounter.Interface.GetMountRefs(target)
+	if err != nil {
+		msg := fmt.Sprintf("failed to check if volume is mounted: %v", err)
+		return nil, status.Error(codes.Internal, msg)
+	}
+
+	// From the spec: If the volume corresponding to the volume_id
+	// is not staged to the staging_target_path, the Plugin MUST
+	// reply 0 OK.
+	if len(refs) == 0 {
```

Comment:
Could it happen that `len(refs) > 1`? And what should we do?

Reply:
If we follow the GetMountRefs code we'll end up [at this function](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/mount/mount_linux.go#L1283-L1290) 
```// This function assumes source cannot be device.
// Some filesystems may share a source name, e.g. tmpfs. And for bind mounting,
// it's possible to mount a non-root path of a filesystem, so we need to use
// root path and major:minor to represent mount source uniquely.
``` 
To answer your question I don't see a way of this happening because we don't support `tmpfs` and afaik that type of bind mounting. So, if `len(refs) > 1` it means `target` is being managed by something else other than this driver. 

I understand the concern with handling that code path but if we return an error the CO will keep retrying. Are you ok with logging or do you prefer to return an error in that scenario?

Reply:
Let's return an error for now

Reply:
Lets log the error for now